proxy/InkAPI.cc

```diff
@@ -9629,3 +9634,35 @@ TSRemapToUrlGet(TSHttpTxn txnp, TSMLoc *urlLocp)
 {
   return remapUrlGet(txnp, urlLocp, &UrlMappingContainer::getToURL);
 }
+
+tsapi char *
+TSHttpTxnClientReqBodyGet(TSHttpTxn txnp, int *len)
```

Comment:
I think we it would be more flexible to return a TSIOBufferReader and do the copying of the bytes in the plugin instead of doing it here.

Using the external APIs in internal code means that either we should be exposing more of the underlying implementation or using the internal APIs directly.

The API could be redone to be something like:
TSIOBufferReader TSIOBufferReaderPostGet() {
  sdk_assert(sdk_sanity_check_txn(txnp) == TS_SUCCESS);
  HttpSM *sm                         = (HttpSM *)txnp;
  return (TSIOBufferReader)sm->get_postbuf_clone_reader();
}

Reply:
Thanks! Sounds reasonable. Will change it.